src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,228 @@
+        String getQuery = "SELECT c.id, c.model, m.id AS manufacturer_id, m.name, m.country"
+                + " FROM cars c"
+                + " JOIN manufacturers m ON c.manufacturer_id = m.id"
+                + " JOIN cars_drivers cd ON c.id = cd.car_id"
+                + " WHERE cd.driver_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarsForDriverStatement =
+                        connection.prepareStatement(getQuery)) {
+            getCarsForDriverStatement.setLong(1, driverId);
+            ResultSet resultSet = getCarsForDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't execute statement "
+                    + "to get cars for driver by driver id " + driverId, e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversForCar(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public List<Driver> getDriversForCar(Long carId) {
+        List<Driver> drivers = new ArrayList<>();
+        String getQuery = "SELECT d.id, d.name, d.licence_number"
+                + " FROM drivers d"
+                + " JOIN cars_drivers cd"
+                + " ON d.id = cd.driver_id WHERE cd.car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getDriversForCarStatement =
+                        connection.prepareStatement(getQuery)) {
+            getDriversForCarStatement.setLong(1, carId);
+            ResultSet resultSet = getDriversForCarStatement.executeQuery();
+            while (resultSet.next()) {
+                drivers.add(getDriverFromResultSet(resultSet));
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't execute statement "
+                    + "to get drivers for car by car id " + carId, e);
+        }
+    }
+
+    private void insertCarAndDriver(Car car) {
+        String createQuery = "INSERT INTO cars_drivers (car_id, driver_id) VALUES(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertCarAndDriverStatement =
+                        connection.prepareStatement(createQuery)) {
+            insertCarAndDriverStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                insertCarAndDriverStatement.setLong(2, driver.getId());
+                insertCarAndDriverStatement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't execute statement "
+                    + "to insert car and it's drivers relationship" + car, e);
+        }
+    }
+
+    private void deleteCarAndDriver(Car car) {
```

Comment:
I do not see any car insert or delete in those methods so let's not name them like they are
```suggestion
    private void insertDriversForCar(Car car) {
        String createQuery = "INSERT INTO cars_drivers (car_id, driver_id) VALUES(?, ?);";
        try (Connection connection = ConnectionUtil.getConnection();
                PreparedStatement insertCarAndDriverStatement =
                        connection.prepareStatement(createQuery)) {
            insertCarAndDriverStatement.setLong(1, car.getId());
            for (Driver driver : car.getDrivers()) {
                insertCarAndDriverStatement.setLong(2, driver.getId());
                insertCarAndDriverStatement.executeUpdate();
            }
        } catch (SQLException e) {
            throw new DataProcessingException("Can't execute statement "
                    + "to insert car and it's drivers relationship" + car, e);
        }
    }

    private void deleteDriversOfCar(Car car) {
```